### PR TITLE
(recommendation): don't list reserved block devices

### PR DESCRIPTION
**Pull request by shovanmaity**
Filter reserved block devices from CStor Pool recommendation. Reserved block devices will have `openebs.io/block-device-tag` label.

### CODE REVIEW

File: common/blockdevice/util.go [hunk 1]
```diff
@@ -256,6 +256,18 @@ func GetDriveType(obj unstructured.Unstructured) (string, error) {
...
 
+// IsReserved checks if block device is resrved or not
+// using openebs.io/block-device-tag label
+func IsReserved(obj unstructured.Unstructured) (bool, error) {
+	if obj.GetKind() != string(types.KindBlockDevice) {
+		return false,
+			errors.Errorf("Can not check block device is reserved or not: Expected kind %q got %q",
+				types.KindBlockDevice, obj.GetKind())
+	}
+	_, ok := obj.GetLabels()["openebs.io/block-device-tag"]
```
Comment (AmitKumarDas) [inline common/blockdevice/util.go:267]: nil pointer check?

Comment (shovanmaity) [reply common/blockdevice/util.go:267]: added test case

Comment (shovanmaity) [reply common/blockdevice/util.go:267]: `GetLabels()` will not return nil map it returns empty map is key not found.

**Review by AmitKumarDas — APPROVED** [review]

File: common/blockdevice/util.go [hunk 2]
```diff
@@ -256,6 +256,18 @@ func GetDriveType(obj unstructured.Unstructured) (string, error) {
 	return unstruct.GetString(&obj, "spec", "details", "driveType")
 }
 
+// IsReserved checks if block device is resrved or not
```
Comment (avishnu) [inline common/blockdevice/util.go:259]: is reserved*

File: common/blockdevice/util.go [hunk 3]
```diff
@@ -299,5 +311,15 @@ func IsEligibleForCStorPool(obj unstructured.Unstructured) (bool, error) {
 		return false, nil
 	}
 
+	// IsReserved checks is block device is reserved or not. If it is reserved
+	// then we can not use that block device to create a CStor pool
+	reserved, err := IsReserved(obj)
+	if err != nil {
+		return false, errors.Wrap(err, "Can not check eligibility for cStor pool.")
```
Comment (avishnu) [inline common/blockdevice/util.go:318]: Cannot

**Pull request merged**